src/components/Home/SearchBar/SearchBar.tsx

```diff
@@ -1,44 +1,58 @@
-import React, { FormEvent, useEffect, useRef, useState } from "react";
+import React, { FC, FormEvent, useEffect, useState } from "react";
 import { getInputFromLocalStorage, saveInputToLocalStorage } from "@/shared/utils/localStorage";
-import { LOCALHOST_INPUT_KEY } from "@/shared/constants";
 import styles from "./SearchBar.module.scss";
+import { IPeople } from "@/interfaces/people.interface";
+import { fetchPeople } from "@/shared/utils/peopleService/fetchPeople";
 
-const SearchBar = () => {
-  const [value, setValue] = useState(getInputFromLocalStorage(LOCALHOST_INPUT_KEY) ?? "");
-  const inputValueRef = useRef(value);
+interface IProps {
+  onSubmitCards: (card: IPeople[]) => void;
+  isLoading: (isLoading: boolean) => void;
+}
 
-  function onUnload(): void {
-    saveInputToLocalStorage(inputValueRef.current);
-  }
+const SearchBar: FC<IProps> = ({ onSubmitCards, isLoading }) => {
+  const [inputValue, setInputValue] = useState(getInputFromLocalStorage());
+  const [searchValue, setSearchValue] = useState(inputValue);
+
+  useEffect(() => {
+    isLoading(true);
+
+    fetchPeople(searchValue)
```

Comment:
Single responsibility principle broken -contains loading logic that is not related to the bar itself. Original behavior of the bar broken.